### PR TITLE
feat: Engagement Center Add advanced filter for realization table to filter by programs - MEED-139 - Meeds-io/MIPs#13 

### PR DESCRIPTION
**Pull request by ArfaouiMohamedAmine**
this change is going to fix the reset option

### CODE REVIEW

File: portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue [hunk 1]
```diff
@@ -114,13 +112,20 @@ export default {
...
+    },
   },
   methods: {
     loadMore() {
       this.retrievePrograms(true ,this.search);
     },
     reset() {
-      this.selected = [];
+      this.selectAll = true;
+      this.selected = this.programsList.map( Object.keys );
```
Comment (AzmiTouil) [inline portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue]:
```suggestion
      this.selected = this.programsList.map(Object.keys);
```

File: portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue [hunk 2]
```diff
@@ -114,13 +112,20 @@ export default {
     selectedPrograms() {
       return  (this.selectAll && !this.partiallySelected && this.search === '') ? [] : this.selected ;
     },
+    emptyList() {
+      return this.selected.length === 0;
+    },
+    partiallySelected () {
+      return this.selected.length !== this.programsList.length;
```
Comment (AzmiTouil) [inline portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue]: should be computed with **hasMore** value also

File: portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue [hunk 3]
```diff
@@ -90,7 +88,7 @@ export default {
   },
   watch: {
     selected() {
-      this.partiallySelected = this.selected.length !== this.programsList.length;
+      this.$emit('empty-list', this.emptyList);
```
Comment (AzmiTouil) [inline portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue]: useless IMO!

**Review by AzmiTouil — CHANGES_REQUESTED** [review]

**Comment by AzmiTouil** [comment]
Reformat code please!

File: portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue (hunk 3 above)
Comment (ArfaouiMohamedAmine) [reply portlets/src/main/webapp/vue-app/realizations/components/FilterRealizationsProgramList.vue]: this value is going to be catched by the parent component 'FilterRealizationsDrawer' to disable the the confirm button whenever the list list is empty !

**Review by AzmiTouil — APPROVED** [review]

**Pull request merged**